### PR TITLE
Phase 1: Server initialisation code factor out. 

### PR DESCRIPTION
**Pull request by jerome-benoit**
+ remove unused attributes; 
+ use inheritance for server configuration tunables; 
+ introduce server types;
+ introduce uws helpers. 

I would like to have that PR merged first, I think I've found a way to avoid to the race condition on WS connection caching on top of it. 

Thanks. 

Signed-off-by: Jérôme Benoit <jerome.benoit@sap.com>

### CODE REVIEW

File: src/server/uWsUtils.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import { App, AppOptions, HttpRequest, HttpResponse, SSLApp, TemplatedApp } from 'uWebSockets.js';
```
Comment (LucasBrazi06) [inline src/server/uWsUtils.ts:1]: Naming of the class uWsUtils.ts -> UWsUtils.ts? :-)

File: src/server/uWsUtils.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      serverName: string, serverModuleName: string, listenCb?: () => void, listen = true): void {
+    let cb: () => void;
+    if (listenCb && typeof listenCb === 'function') {
+      cb = listenCb;
+    } else {
+      cb = async () => {
+        await ServerUtils.defaultListenCb(serverModuleName, 'startServer', serverName, serverConfig.protocol, serverConfig.host ?? '::', serverConfig.port);
+      };
+    }
+
```
Comment (LucasBrazi06) [inline src/server/uWsUtils.ts:37]: BL

**Review by LucasBrazi06 — APPROVED** [review]
GUT!
Tu t'es bien amusé ;-)

File: src/server/uWsUtils.ts (hunk 1 above)
Comment (jerome-benoit) [reply src/server/uWsUtils.ts:1]: The official short name of the library is uWs, I've just reused it.

Comment (LucasBrazi06) [reply src/server/uWsUtils.ts:1]: > The official short name of the library is uWs, I've just reused it.

I know but here this is a class name and the lib doesn't have any classes that start with a minus letter with what I see in the import ;-)

**Pull request merged**